### PR TITLE
Tidy update semaphore v2 build, fix caching issue

### PR DESCRIPTION
**Pull request by Brian-McM**
## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -9,44 +9,37 @@ agent:
...
     task:
       jobs:
         - name: Build
           commands:
-            - checkout
-            - cache restore go-pkg-cache
-            - cache restore go-mod-cache
+            - make build
             - 'cache store bin-${SEMAPHORE_GIT_SHA} bin'
             - cache store go-pkg-cache .go-pkg-cache
```
Comment (bcreane) [inline .semaphore/semaphore.yml:29]: are the single-quotes around some of the `cache` commands required for macro/variable expansion?

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -9,44 +9,37 @@ agent:
     type: e1-standard-2
     os_image: ubuntu1804
 
+global_job_config:
```
Comment (bcreane) [inline .semaphore/semaphore.yml:12]: we discussed the fact that none of the blocks (other than "Build") execute a `cache store`, however, currently all other blocks depend (either directly or indirectly) upon the "Build" block.
@Brian-McM - wdyt about adding a comment connecting the `cache store` commands in "Build" block to the prologue's actions, if you think it's helpful.

File: .semaphore/semaphore.yml (hunk 1 above)
Comment (Brian-McM) [reply .semaphore/semaphore.yml:29]: Yes it's for the ${SEMAPHORE_GIT_SHA} expansion

File: .semaphore/semaphore.yml (hunk 2 above)
Comment (Brian-McM) [reply .semaphore/semaphore.yml:12]: I think that's a good idea, I'll add the comment.

**Pull request merged**